pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -323,9 +343,23 @@ func (r ReplicaSetReconciler) buildAutomationConfigConfigMap(mdb mdbv1.MongoDB)
 	if err != nil {
 		return corev1.ConfigMap{}, fmt.Errorf("error reading version manifest from disk: %+v", err)
 	}
-	ac := buildAutomationConfig(mdb, manifest.BuildsForVersion(mdb.Spec.Version))
-	acBytes, err := json.Marshal(ac)
-	if err != nil {
+	currentCm := corev1.ConfigMap{}
+	currentAc := automationconfig.AutomationConfig{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: mdb.ConfigMapName(), Namespace: mdb.Namespace}, &currentCm); err != nil {
+		if !errors.IsNotFound(err) {
+			return corev1.ConfigMap{}, err
+		}
+	} else {
+		if err := json.Unmarshal([]byte(currentCm.Data[AutomationConfigKey]), &currentAc); err != nil {
+			return corev1.ConfigMap{}, err
+		}
+	}
```

Comment:
we don't actually need this to be in an `else` clause. (another thing that keeps the happy path as left as possible)